gordo_components/model/anomaly/diff.py

```diff
@@ -78,6 +79,111 @@ def fit(self, X: np.ndarray, y: np.ndarray):
         self.scaler.fit(y)  # Scaler is used for calculating errors in .anomaly()
         return self
 
+    def cross_validate(
+        self,
+        *,
+        X: Union[pd.DataFrame, np.ndarray],
+        y: Optional[Union[pd.DataFrame, np.ndarray]] = None,
+        cv=TimeSeriesSplit(n_splits=3),
+        **kwargs,
+    ):
+        """
+        Run cross validation on the model, and will update the model's threshold value based
+        on the cross validation folds
+
+        Parameters
+        ----------
+        X: Union[pd.DataFrame, np.ndarray]
+            Input data to the model
+        y: Optional[Union[pd.DataFrame, np.ndarray]]
+            Target data
+        kwargs: dict
+            Any additional kwargs to be passed to :func:`sklearn.model_selection.cross_validate`
+
+        Returns
+        -------
+        dict
+        """
+        # Depend on having the trained fold models
+        kwargs.update(dict(return_estimator=True, cv=cv))
+
+        X = X if not isinstance(X, pd.DataFrame) else X.values
```

Comment:
Too many levels of negation for my head, I find `X = X.values if isinstance(X, pd.DataFrame) else X` much easier to read. No?